src/levels.rs

```diff
@@ -390,6 +391,18 @@ impl Neg for Mv {
     }
 }
 
+impl PartialEq for Mv {
+    #[inline(always)]
+    fn eq(&self, other: &Self) -> bool {
+        // In C, `mv` is a union of either two int16_t values or a uint32_t,
+        // with the uint32_t variant used for comparisons as it is faster.
+        let this: u32 = transmute!(*self);
+        let other: u32 = transmute!(*other);
+
+        this == other
```

Comment:
```suggestion
        self.as_bytes() == other.as_bytes()
```

Same for the others, as @daxtens suggested.

I'll think about what we should leave for a comment.  I think you can remove the comments you left for now, as the issue is a bit more generic and subtle than that.

Reply:
@kkysen done, let me know how you want to document this. Maybe "Bytewise comparison optimizes better than per-field comparison"?

Reply:
It seems to be mostly about not comparing per-field with `&&`.  I'll give a suggestion in a new comment.